### PR TITLE
explorer: barLoad is checked for nil already.

### LINKED ISSUE #1103: data race
commit fcd8c3488fd393703274587dd2ca1cf8582dd599 
```
==================                                                                                                   
WARNING: DATA RACE                                                                                                  
Write at 0x00c0239d6078 by goroutine 100:                                                                            
  github.com/decred/dcrdata/v4/explorer.(*explorerUI).BeginSyncStatusUpdates.func2.1()                              
      /home/dcrdata/git/dcrdata/explorer/syncstatus.go:92 +0xca                                                      
  github.com/decred/dcrdata/v4/explorer.(*explorerUI).BeginSyncStatusUpdates.func2()                                
      /home/dcrdata/git/dcrdata/explorer/syncstatus.go:103 +0x926                                                    
                                                                                                                    
Previous read at 0x00c0239d6078 by goroutine 58:                                                                     
  github.com/decred/dcrdata/v4/explorer.(*explorerUI).BeginSyncStatusUpdates.func1()                                
      /home/dcrdata/git/dcrdata/explorer/syncstatus.go:74 +0x171                                                     
                                                                                                                    
Goroutine 100 (running) created at:                                                                                  
  github.com/decred/dcrdata/v4/explorer.(*explorerUI).BeginSyncStatusUpdates()                                      
      /home/dcrdata/git/dcrdata/explorer/syncstatus.go:86 +0xea                                                      
  main._main()                                    
      /home/dcrdata/git/dcrdata/main.go:595 +0x2797                                                                  
  github.com/decred/dcrdata/v4/db/dcrpg.NewChainDBWithCancel()                                                      
      /home/dcrdata/git/dcrdata/db/dcrpg/pgblockchain.go:446 +0x18e                                                 
  main._main()                                                                                                      
      /home/dcrdata/git/dcrdata/main.go:187 +0xabc7                                                                 
  fmt.Sscanf()                                                                                                      
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0                                                                     
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()                                                       
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144       
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279       
  fmt.Sscanf()                                                                                                      
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0                                                                    
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()                                                       
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144      
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()                                               
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279      
  github.com/btcsuite/goleveldb/leveldb.(*DB).checkAndCleanFiles()                                                  
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db_util.go:52 +0x2e4                    
  github.com/btcsuite/goleveldb/leveldb.openDB()                                                                    
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:130 +0xa00                        
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()                                               
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279      
  fmt.Sscanf()                                                     
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0                                                                    
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()                                                       
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144      
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()            
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279      
  fmt.Sscanf()                                                                   
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0                                                                    
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()                                                       
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144      
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()                                               
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279      
  fmt.Sscanf()                                                                        
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0                                                                    
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()                                                       
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:657 +0x1ed       
  fmt.Fscanf()                                                                                                      
      /home/dcrdata/go/src/fmt/scan.go:143 +0xe9                                                                    
  fmt.Sscanf()                                                                                                      
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0                                                                    
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()                    
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144      
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  github.com/btcsuite/goleveldb/leveldb.(*DB).recoverJournal()                                              [62/1833]
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:476 +0xe3
  github.com/btcsuite/goleveldb/leveldb.openDB()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:125 +0x9dc
  github.com/btcsuite/goleveldb/leveldb.Open()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:197 +0x1a0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseNamePtr()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:666 +0x46
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).GetMeta.func1()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:332 +0x42d
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).GetMeta.func2()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:356 +0xc9
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).GetMeta()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:407 +0x775
  github.com/btcsuite/goleveldb/leveldb.(*session).recover()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/session.go:119 +0xde
  github.com/btcsuite/goleveldb/leveldb.Open()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:183 +0xd1
  github.com/btcsuite/goleveldb/leveldb.OpenFile()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:219 +0xa5
  github.com/decred/dcrd/database/ffldb.openDB()
      /home/dcrdata/go/pkg/mod/github.com/decred/dcrd/database@v1.0.3/ffldb/db.go:2068 +0x183
  github.com/decred/dcrd/database/ffldb.openDBDriver()
      /home/dcrdata/go/pkg/mod/github.com/decred/dcrd/database@v1.0.3/ffldb/driver.go:53 +0xf7
  github.com/decred/dcrd/database.Open()
      /home/dcrdata/go/pkg/mod/github.com/decred/dcrd/database@v1.0.3/driver.go:89 +0x105
  github.com/decred/dcrdata/v4/stakedb.(*StakeDatabase).Open()
      /home/dcrdata/git/dcrdata/stakedb/stakedb.go:737 +0x21c
  github.com/decred/dcrdata/v4/stakedb.NewStakeDatabase()
      /home/dcrdata/git/dcrdata/stakedb/stakedb.go:264 +0x98f
  github.com/decred/dcrdata/v4/db/dcrsqlite.newWiredDB()
      /home/dcrdata/git/dcrdata/db/dcrsqlite/apisource.go:62 +0x271
  github.com/decred/dcrdata/v4/db/dcrsqlite.InitWiredDB()
      /home/dcrdata/git/dcrdata/db/dcrsqlite/apisource.go:107 +0xa5
  main._main()
      /home/dcrdata/git/dcrdata/main.go:151 +0xfd0
  main.main()
      /home/dcrdata/git/dcrdata/main.go:56 +0x9e

Goroutine 58 (running) created at:
  github.com/decred/dcrdata/v4/explorer.(*explorerUI).BeginSyncStatusUpdates()
      /home/dcrdata/git/dcrdata/explorer/syncstatus.go:71 +0xbe
  main._main()
      /home/dcrdata/git/dcrdata/main.go:595 +0x2797
  github.com/decred/dcrdata/v4/db/dcrpg.NewChainDBWithCancel()
      /home/dcrdata/git/dcrdata/db/dcrpg/pgblockchain.go:446 +0x18e
  main._main()
      /home/dcrdata/git/dcrdata/main.go:187 +0xabc7
  fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  github.com/btcsuite/goleveldb/leveldb.(*DB).checkAndCleanFiles()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db_util.go:52 +0x2e4
  github.com/btcsuite/goleveldb/leveldb.openDB()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:130 +0xa00
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
 fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:657 +0x1ed
  fmt.Fscanf()
      /home/dcrdata/go/src/fmt/scan.go:143 +0xe9
  fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  fmt.Sscanf()
      /home/dcrdata/go/src/fmt/scan.go:114 +0xd0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseName()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:643 +0x144
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:458 +0x279
  github.com/btcsuite/goleveldb/leveldb.(*DB).recoverJournal()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:476 +0xe3
  github.com/btcsuite/goleveldb/leveldb.openDB()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:125 +0x9dc
  github.com/btcsuite/goleveldb/leveldb.Open()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:197 +0x1a0
  github.com/btcsuite/goleveldb/leveldb/storage.fsParseNamePtr()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:666 +0x46
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).GetMeta.func1()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:332 +0x42d
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).GetMeta.func2()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:356 +0xc9
  github.com/btcsuite/goleveldb/leveldb/storage.(*fileStorage).GetMeta()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/storage/file_storage.go:407 +0x775
  github.com/btcsuite/goleveldb/leveldb.(*session).recover()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/session.go:119 +0xde
  github.com/btcsuite/goleveldb/leveldb.Open()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:183 +0xd1
  github.com/btcsuite/goleveldb/leveldb.OpenFile()
      /home/dcrdata/go/pkg/mod/github.com/btcsuite/goleveldb@v1.0.0/leveldb/db.go:219 +0xa5
  github.com/decred/dcrd/database/ffldb.openDB()
      /home/dcrdata/go/pkg/mod/github.com/decred/dcrd/database@v1.0.3/ffldb/db.go:2068 +0x183
  github.com/decred/dcrd/database/ffldb.openDBDriver()
      /home/dcrdata/go/pkg/mod/github.com/decred/dcrd/database@v1.0.3/ffldb/driver.go:53 +0xf7
  github.com/decred/dcrd/database.Open()
      /home/dcrdata/go/pkg/mod/github.com/decred/dcrd/database@v1.0.3/driver.go:89 +0x105
  github.com/decred/dcrdata/v4/stakedb.(*StakeDatabase).Open()
      /home/dcrdata/git/dcrdata/stakedb/stakedb.go:737 +0x21c
  github.com/decred/dcrdata/v4/stakedb.NewStakeDatabase()
      /home/dcrdata/git/dcrdata/stakedb/stakedb.go:264 +0x98f
  github.com/decred/dcrdata/v4/db/dcrsqlite.newWiredDB()
      /home/dcrdata/git/dcrdata/db/dcrsqlite/apisource.go:62 +0x271
  github.com/decred/dcrdata/v4/db/dcrsqlite.InitWiredDB()
      /home/dcrdata/git/dcrdata/db/dcrsqlite/apisource.go:107 +0xa5
  main._main()
      /home/dcrdata/git/dcrdata/main.go:151 +0xfd0
  main.main()
      /home/dcrdata/git/dcrdata/main.go:56 +0x9e
==================

### PR DESCRIPTION
**Pull request by dajohi**
Fixes #1103

### CODE REVIEW

File: explorer/syncstatus.go [hunk 1]
```diff
@@ -71,11 +71,6 @@ func (exp *explorerUI) BeginSyncStatusUpdates(barLoad chan *dbtypes.ProgressBarL
 	go func() {
 		timer := time.NewTicker(syncStatusInterval)
 		for range timer.C {
-			if barLoad == nil {
-				log.Debug("Stopping progress bar signals.")
-				timer.Stop()
```
Comment (chappjc) [inline explorer/syncstatus.go]: Another solution to avoid leaking this goroutine is needed.  Note that the other goroutine below nils `barLoad`.

Comment (chappjc) [reply explorer/syncstatus.go]: Maybe just stop the timer in the other goroutine. ofc would need to capture the timer by both goroutines.

Comment (dmigwi) [reply explorer/syncstatus.go]: Since the timer never stops the goroutine will always be running in the background.

Comment (dajohi) [reply explorer/syncstatus.go]: pushed a revision.

**Comment by dajohi** [comment]
Closed by #1106

**Pull request closed without merging**